---uppsrc/ide/Makefile.orig	Mon Mar 20 18:36:35 2006
+++ uppsrc/ide/Makefile	Mon Mar 20 18:37:40 2006
@@ -1,13 +1,13 @@
-UPPDIR1 = /src/uppsrc/
+UPPDIR1 = ../
 
-UPPOUT = /home/cxl/upp/out/
-CINC =  -I$(UPPDIR1)
-Macro =  -DflagGUI -DflagGCC -DflagSHARED -DflagLINUX
-CC = c++ -c -O2 -Os -ffunction-sections 
-C_FLAGS = $(CC) -x c
-CPPFLAGS = $(CC) -x c++ 
-LIBPATH =  -L"/usr/X11R6/lib"
+UPPOUT = $(UPPDIR1)../out/
+CINC =  -I$(UPPDIR1) -I$(X11BASE)/include -I$(LOCALBASE)/include -I$(LOCALBASE)/include/freetype2
+Macro =  -DflagGUI -DflagGCC -DflagSHARED -DflagFREEBSD
+C_FLAGS = $(CC) -c $(CFLAGS)
+CPPFLAGS = $(CXX) -c $(CXXFLAGS)
+LIBPATH =  -L$(X11BASE)/lib -L$(LOCALBASE)/lib
 AR = ar -sr
+MKDIR?=	mkdir -p
 OutDir_ide = $(UPPOUT)ide/GCC32-Gcc-Gui-Linux-Main-Shared/
 Macro_ide = $(Macro) -DflagMAIN
 OutDir_ide_Common = $(UPPOUT)ide/Common/GCC32-Gcc-Gui-Linux-Shared/
@@ -76,47 +76,47 @@
 Macro_PdfDraw = $(Macro)
 
 OutDir = $(OutDir_ide)
-OutFile = /src/ide
+OutFile = $(UPPDIR1)../ide
 
 .PHONY: all
 all: install $(OutFile)
 
 .PHONY: install
 install:
-	-mkdir -p $(OutDir)
-	-mkdir -p $(OutDir_ide)
-	-mkdir -p $(OutDir_ide_Common)
-	-mkdir -p $(OutDir_ide_LayDes)
-	-mkdir -p $(OutDir_ide_IconDes)
-	-mkdir -p $(OutDir_ide_Builders)
-	-mkdir -p $(OutDir_ide_Debuggers)
-	-mkdir -p $(OutDir_ide_Browser)
-	-mkdir -p $(OutDir_ide_Topic)
-	-mkdir -p $(OutDir_ide_VectorDes)
-	-mkdir -p $(OutDir_CodeEditor)
-	-mkdir -p $(OutDir_Topic)
-	-mkdir -p $(OutDir_CtrlLib)
-	-mkdir -p $(OutDir_Web)
-	-mkdir -p $(OutDir_plugin_bz2)
-	-mkdir -p $(OutDir_HexView)
-	-mkdir -p $(OutDir_Esc)
-	-mkdir -p $(OutDir_Core)
-	-mkdir -p $(OutDir_RichEdit)
-	-mkdir -p $(OutDir_plugin_png)
-	-mkdir -p $(OutDir_plugin_gif)
-	-mkdir -p $(OutDir_Image)
-	-mkdir -p $(OutDir_plugin_tif)
-	-mkdir -p $(OutDir_plugin_jpg)
-	-mkdir -p $(OutDir_Draw)
-	-mkdir -p $(OutDir_plugin_z)
-	-mkdir -p $(OutDir_coff_binobj)
-	-mkdir -p $(OutDir_coff)
-	-mkdir -p $(OutDir_plugin_ndisasm)
-	-mkdir -p $(OutDir_CppBase)
-	-mkdir -p $(OutDir_VectorDes)
-	-mkdir -p $(OutDir_RichText)
-	-mkdir -p $(OutDir_CtrlCore)
-	-mkdir -p $(OutDir_PdfDraw)
+	-$(MKDIR) $(OutDir)
+	-$(MKDIR) $(OutDir_ide)
+	-$(MKDIR) $(OutDir_ide_Common)
+	-$(MKDIR) $(OutDir_ide_LayDes)
+	-$(MKDIR) $(OutDir_ide_IconDes)
+	-$(MKDIR) $(OutDir_ide_Builders)
+	-$(MKDIR) $(OutDir_ide_Debuggers)
+	-$(MKDIR) $(OutDir_ide_Browser)
+	-$(MKDIR) $(OutDir_ide_Topic)
+	-$(MKDIR) $(OutDir_ide_VectorDes)
+	-$(MKDIR) $(OutDir_CodeEditor)
+	-$(MKDIR) $(OutDir_Topic)
+	-$(MKDIR) $(OutDir_CtrlLib)
+	-$(MKDIR) $(OutDir_Web)
+	-$(MKDIR) $(OutDir_plugin_bz2)
+	-$(MKDIR) $(OutDir_HexView)
+	-$(MKDIR) $(OutDir_Esc)
+	-$(MKDIR) $(OutDir_Core)
+	-$(MKDIR) $(OutDir_RichEdit)
+	-$(MKDIR) $(OutDir_plugin_png)
+	-$(MKDIR) $(OutDir_plugin_gif)
+	-$(MKDIR) $(OutDir_Image)
+	-$(MKDIR) $(OutDir_plugin_tif)
+	-$(MKDIR) $(OutDir_plugin_jpg)
+	-$(MKDIR) $(OutDir_Draw)
+	-$(MKDIR) $(OutDir_plugin_z)
+	-$(MKDIR) $(OutDir_coff_binobj)
+	-$(MKDIR) $(OutDir_coff)
+	-$(MKDIR) $(OutDir_plugin_ndisasm)
+	-$(MKDIR) $(OutDir_CppBase)
+	-$(MKDIR) $(OutDir_VectorDes)
+	-$(MKDIR) $(OutDir_RichText)
+	-$(MKDIR) $(OutDir_CtrlCore)
+	-$(MKDIR) $(OutDir_PdfDraw)
 
 $(OutFile):  \
 	$(OutDir_ide)SelectPkg.o \
@@ -182,7 +182,7 @@
 	$(OutDir_RichText)RichText.a \
 	$(OutDir_CtrlCore)CtrlCore.a \
 	$(OutDir_PdfDraw)PdfDraw.a
-	c++ -o $(OutFile) -Wl,-s $(LIBPATH) -Wl,-O,2 $(LINKOPTIONS) \
+	$(CXX) -o $(OutFile) -Wl,-s $(LIBPATH) -Wl,-O,2 $(LINKOPTIONS) \
 		$(OutDir_ide)SelectPkg.o \
 		$(OutDir_ide)UppWspc.o \
 		$(OutDir_ide)UppDlg.o \
